siocli/lib/SqlDump.cpp

```diff
@@ -0,0 +1,771 @@
+
+    return data;
+}
+
+std::uint8_t readUint16(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint16_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read int16 failed");
+
+    return data;
+}
+
+std::uint32_t readUint32(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint32_t data = 0;
+    if (!input.ReadVarint32(&data)) throw std::runtime_error("Read int32 failed");
+
+    return data;
+}
+
+std::uint64_t readUint64(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint64_t data = 0;
+    if (!input.ReadVarint64(&data)) throw std::runtime_error("Read int64 failed");
+
+    return data;
+}
+
+float readFloat(google::protobuf::io::CodedInputStream& input)
+{
+    float data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read float failed");
+
+    return data;
+}
+
+double readDouble(google::protobuf::io::CodedInputStream& input)
+{
+    double data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read double failed");
+
+    return data;
+}
+
+bool readBool(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint8_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read bool failed");
+
+    return data > 0;
+}
+
+std::string readValue(google::protobuf::io::CodedInputStream& input, ColumnDataType columnDataType)
+{
+    switch (columnDataType) {
+        case ColumnDataType::COLUMN_DATA_TYPE_BOOL: {
+            return readBool(input) ? "true" : "false";
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_INT8: {
+            return std::to_string(readInt8(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_UINT8: {
+            return std::to_string(readUint8(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_INT16: {
+            return std::to_string(readInt16(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_UINT16: {
+            return std::to_string(readUint16(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_INT32: {
+            return std::to_string(readInt32(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_UINT32: {
+            return std::to_string(readUint32(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_INT64: {
+            return std::to_string(readInt64(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_UINT64: {
+            return std::to_string(readUint64(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_FLOAT: {
+            return std::to_string(readFloat(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_DOUBLE: {
+            return std::to_string(readDouble(input));
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_TEXT: {
+            std::stringstream ss;
+            ss << '\'' << readString(input) << '\'';
```

Comment:
You must process string char by char and emit quotes as needed.

Reply:
Could you please clarify what do you mean?

I see some code around _case siodb::COLUMN_DATA_TYPE_TEXT:_ in **Client.cpp** I have to do the same?

Reply:
No, not the same. You should only escape:
- single quote `'` as `''`
- '\n' (0xA) as `+ TO_CHAR(10) +` (break string, add '+' operator, continue string)
- '\r' (0xD) as `+ TO_CHAR(13) +` (break string, add '+' operator, continue string)